openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,839 @@
-                        scaleCIvalues.push( (tempCIvalues[j] - tempCImin) / (tempCImax - tempCImin) );
-                    }
-
-                    // Call the applyScaledValues method
-                    for (var l in scaledCatIndicator) {
-                        scaledCatIndicator[l][CIkeys[i]] = scaledCatIndicator[l].scaleCIvalues(l);
+                indicatorString.push(munic + '|'+ theme +'|'+ tempSum);
+            } else if ( operator == "Weighted multiplication") {
+                for (var p4 in la[o].properties) {
+                    // iterate over the indicator child keys
+                    for (var r4 = 0; r4 < tempIndicatorChildrenKeys.length; r4++) {
+                        if (p4 == tempIndicatorChildrenKeys[r4]) {
+                            // Sum the theme indicators
+                            var weight = tempChildren[r4].weight;
+                            if (tempSum == 0) {
+                                tempSum = (la[o].properties[p4] * weight);
+                            } else {
+                                tempSum = tempSum * (la[o].properties[p4] * weight);
+                            }
+                        }
                     }
                 }
-                scaleCIvalues = [];
-                tempCIvalues = [];
-            }
-
-            // The data passed into d3 need to be an array of objects
-            // Place the category indicator objects into an array
-            var catData = [];
-
-            for (var k in scaledCatIndicator) {
-                catData.push(scaledCatIndicator[k]);
-            }
-
-            // TODO remove this concat if need be
-            var concat = catData.concat(previousCatData);
-
-            Category_PCP_Chart(catData, municipality, districName, concat);
-
-            for (var k in scaledCatIndicator) {
-                previousCatData.push(scaledCatIndicator[k]);
-            }
-
-            /////////////////////////////////////////////
-            /////////// Create the svi object ///////////
-            /////////////////////////////////////////////
-
-            for (var k in sviIndicator) {
-                delete sviIndicator[k];
-            }
-
-            for (var k in scaledSviIndicator) {
-                delete scaledSviIndicator[k];
-            }
-
-            for (var k in weightedSviIndicator) {
-                delete weightedSviIndicator[k];
-            }
-
-            function newSVIvalues(SVIaverage, tempMunic) {
-                sviIndicator[tempMunic] = SVIaverage;
-            }
-
-            // Associate the newSVIvalues method with the svi object
-            sviIndicator.SVIvalues = newSVIvalues;
-            var tempSviParentChildKey = [];
-            var tempSviIndicator = {};
-            for (var i = 0; i < municipality.length; i++) {
-                tempSviIndicator[municipality[i]] = [];
+                indicatorString.push(munic + '|'+ theme +'|'+ tempSum);
```

Comment:
If you had an array such as indicatorInfo = [],
then you can do something like
indicatorData.push({"region": region, "theme": theme, "sum": sum})
And afterwards you can access it easily without parsing a string.